server/src/main/java/org/opensearch/index/mapper/DateFieldMapper.java

```diff
@@ -463,10 +467,10 @@ public Query rangeQuery(
             }
             DateMathParser parser = forcedDateParser == null ? dateMathParser : forcedDateParser;
             return dateRangeQuery(lowerTerm, upperTerm, includeLower, includeUpper, timeZone, parser, context, resolution, (l, u) -> {
+                Query pointRangeQuery = createPointRangeQuery(l, u);
```

Comment:
We're creating the `ApproximatePointRangeQuery` even if `isSearchable()` is not true.

Maybe make this:

```
Query pointRangeQuery = isSearchable() ? createPointRangeQuery(l, u) : null;
```

for symmetry with the DV case?